src/decisionengine/framework/util/tests/test_metrics.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def env_setup(tmp_path, monkeypatch):
+    """Make sure we have a directory set for PROMETHEUS_MULTIPROC_DIR so that
+    metric instantiation gives us multiprocess metrics"""
+    d = tmp_path
+    monkeypatch.setenv("PROMETHEUS_MULTIPROC_DIR", str(d))
+    yield
+    monkeypatch.delenv("PROMETHEUS_MULTIPROC_DIR")
+
+
+# Put these metrics declarations in fixtures because we need the env_setup
+# fixture to be set before importing any of the metrics
+@pytest.fixture()
+def Gauge():
+    from decisionengine.framework.util.metrics import Gauge
```

Comment:
I'm wondering if we can solve the tmp_path with https://docs.python.org/3/library/importlib.html#importlib.reload  In theory that should let is change out the `PROMETHEUS_MULTIPROC_DIR` with each invocation

Reply:
I can certainly play a bit with it right now to see I can solve it.

Reply:
It didn't work for me.  I think it's because of this caveat from the page you linked:

"Other references to the old objects (such as names external to the module) are not rebound to refer to the new objects and must be updated in each namespace where they occur if that is desired."

Since `prometheus_client.values.ValueClass` is set at the first import, and I'm reloading the API `decisionengine.framework.util.metrics` at each Gauge invocation, `ValueClass` is untouched, and is thus referring to the wrong class.

Reply:
Yeah, that sounds yucky....

Reply:
Alright - so I'll just leave it as is then.